crossbeam-channel/benchmarks/plot.py

```diff
@@ -17,68 +17,44 @@
 def plot(subplot, title, prefix, runs):
     runs.reverse()
 
-    ys = [6 * (i+1) for i in xrange(len(runs))]
+    ys = [6 * (i + 1) for i in range(len(runs))]
     ax = fig.add_subplot(subplot)
     ax.set_title(title)
     ax.set_yticks(ys)
     ax.set_yticklabels(runs)
     ax.tick_params(which='major', length=0)
     ax.set_xlabel('seconds')
 
-    go = [0] * len(runs)
-    mpsc = [0] * len(runs)
-    futures_channel = [0] * len(runs)
-    chan = [0] * len(runs)
-    crossbeam_channel = [0] * len(runs)
+    scores = {}
 
     for (i, run) in enumerate(runs):
         for (test, lang, impl, secs) in results:
             if test == prefix + '_' + run:
-                if lang == 'Go' and impl == 'chan':
-                    go[i] = secs
-                if lang == 'Rust' and impl == 'mpsc':
-                    mpsc[i] = secs
-                if lang == 'Rust' and impl == 'futures-channel':
-                    futures_channel[i] = secs
-                if lang == 'Rust' and impl == 'chan':
-                    chan[i] = secs
-                if lang == 'Rust' and impl == 'crossbeam-channel':
-                    crossbeam_channel[i] = secs
+                name = lang + '_' + impl
+                if name not in scores:
+                    scores[name] = [0] * len(runs)
+                scores[name][i] = secs
 
     opts = dict(height=0.7, align='center')
-    ax.barh([y-2 for y in ys], go, color='skyblue', **opts)
-    ax.barh([y-1 for y in ys], crossbeam_channel, color='red', **opts)
-    ax.barh([y+0 for y in ys], chan, color='orange', **opts)
-    ax.barh([y+1 for y in ys], mpsc, color='black', **opts)
-    ax.barh([y+2 for y in ys], futures_channel, color='blue', **opts)
+    for (i, score) in enumerate(scores.values()):
+        ax.barh([y + i - len(scores) // 2 for y in ys], score, **opts)
```

Comment:
We're losing the different colors here, perhaps have a fixed `colors` list that you index with `i`?

Reply:
Matplotlib will allocate different colors automatically